x-pack/plugins/observability/public/components/shared/alert_search_bar/alert_search_bar_with_url_sync.tsx

```diff
@@ -12,13 +12,21 @@ import {
   useAlertSearchBarStateContainer,
 } from './containers';
 import { ObservabilityAlertSearchBar } from './alert_search_bar';
+import { ObservabilityAlertSearchBarProvider } from './services';
 import { AlertSearchBarWithUrlSyncProps } from './types';
+import { useKibana } from '../../../utils/kibana_react';
+import { ObservabilityAppServices } from '../../../application/types';
 
 function AlertSearchbarWithUrlSync(props: AlertSearchBarWithUrlSyncProps) {
   const { urlStorageKey, ...searchBarProps } = props;
   const stateProps = useAlertSearchBarStateContainer(urlStorageKey);
+  const services = useKibana<ObservabilityAppServices>().services;
```

Comment:
Take every opportunity to eliminate `useKibana`.  If you move the Provider to your React root, these lines get deleted.

Reply:
Wondering about the reasoning for it, `AlertSearchbarWithUrlSync` is an internal component (only used within the Observability plugin) and I put the provider here to make `AlertSearchbarWithUrlSync` stand alone without a need to always wrap the component with this provider on every usage. Will it cause an issue?

Reply:
To make a component shareable _and_ portable, we need to prioritize decoupling it from Kibana.  The `useKibana` hook is incredibly troublesome, as it literally allows a plugin to slam entire start contracts into context, all without any checks except at runtime.

In short: you have no idea if `ObservabilityAppServices` is actually _in_ context when this component is consumed elsewhere.  It's an implementation detail a consumer _must_ know, and due to the nature of `useKibana`, they _won't know_ until runtime.  It's a constant problem.

This is why creating a provider for the shared component and putting it in the root is so helpful-- it allows us to provide and enforce dependency injection.

Reply:
> you have no idea if ObservabilityAppServices is actually in context when this component is consumed elsewhere.

Since this component is meant to be used inside of the `Observability` plugin, it can assume that `ObservabilityAppServices` is provided through `useKibana`. I understand using these shared components between plugins can cause an issue, hence the creation of the related provider, but I don't see the issue that you mentioned in sharing components inside of a plugin. Am I missing something?

>This is why creating a provider for the shared component and putting it in the root is so helpful-- it allows us to provide and enforce dependency injection.

How do we do that at the root level?